src/main/java/no/unit/nva/cristin/projects/CristinApiClient.java

```diff
@@ -120,7 +122,7 @@ public ProjectsWrapper queryCristinProjectsIntoWrapperObjectWithAdditionalMetada
         // TODO: NP-2385: Use Link header / Pagination data from Cristin response in the next two values
         projectsWrapper.setFirstRecord(0);
         projectsWrapper.setNextResults(null); // TODO: Change to URI
-        projectsWrapper.setHits(transformCristinProjectsToNvaProjects(enrichedProjects));
+        projectsWrapper.setHits(transformToNvaProject(enrichedProjects));
```

Comment:
Suggestion:
In  CrstinApiClient:
```
  public ProjectsWrapper queryCristinProjectsIntoWrapperObjectWithAdditionalMetadata(
        Map<String, String> requestQueryParams)
        throws IOException, URISyntaxException {

        long startRequestTime = System.currentTimeMillis();

        long endRequestTime = System.currentTimeMillis();

        List<NvaProject> nvaProjects = importProjectsFromCristin(requestQueryParams);
        ProjectsWrapper projectsWrapper = new ProjectsWrapper();

        String requestQueryParamsAsString = String.format(TITLE_AND_LANGUAGE_QUERY_PARAM_PLACEHOLDER,
            requestQueryParams.get(TITLE), requestQueryParams.get(LANGUAGE));

        projectsWrapper.setId(buildUri(BASE_URL, QUESTION_MARK + requestQueryParamsAsString));
        projectsWrapper.setSize(0); // TODO: NP-2385: X-Total-Count header from Cristin response
        projectsWrapper.setSearchString(requestQueryParamsAsString);
        projectsWrapper.setProcessingTime(calculateProcessingTime(startRequestTime, endRequestTime));
        // TODO: NP-2385: Use Link header / Pagination data from Cristin response in the next two values
        projectsWrapper.setFirstRecord(0);
        projectsWrapper.setNextResults(null); // TODO: Change to URI
        projectsWrapper.setHits(nvaProjects);

        return projectsWrapper;
    }

    private List<NvaProject> importProjectsFromCristin(Map<String, String> requestQueryParams) throws IOException, URISyntaxException {
        return queryAndEnrichProjects(requestQueryParams)
            .filter(CristinProject::hasValidContent)
                  .map(CristinProject::toNvaProject)
                  .collect(Collectors.toList());
    }
```
In CristinProject:

```
  public NvaProject toNvaProject(){
        return new NvaProjectBuilder(this).build();
    }
```

Reply:
I notice that we collect a stream and we stream it again so basically we do this:
```
someList.stream().map(x->toY(x)).collect().stream().map(y->toZ(z)).collect().stream()...
```
If a stream is to be processed once, it is better to collect only at the end I think.  I find no harm to break the processing of stream in several parts but we do not need to collect and stream again if there is no reason for that.
In my above suggestion `queryAndEnrichProjects` returns a Stream.